src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStreamReadAsyncMemoryOverloadsTests.cs

```diff
@@ -0,0 +1,558 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+
+namespace Microsoft.NetCore.Analyzers.Runtime.UnitTests
+{
+    public class PreferStreamReadAsyncMemoryOverloadsTest : PreferStreamAsyncMemoryOverloadsTestBase
+    {
+        #region C# - No diagnostic - Analyzer
+
+        [Fact]
+        public async Task CS_Analyzer_NoDiagnostic_Read()
+        {
+            await AnalyzeCSAsync(@"
```

Comment:
Theoretically none of these tests need to be async/await.. they can all just return the underlying call.

e.g.

```suggestion
        public Task CS_Analyzer_NoDiagnostic_Read()
        {
            return AnalyzeCSAsync(@"
```